### PR TITLE
add openweather backend

### PR DESCRIPTION
**Pull request by R11baka**
Add openweather backend with 5 day forecast

### CODE REVIEW

**Comment by schachmat** [comment]
Heyho @R11baka,

Thanks for your work! Please resolve the merge conflicts and remove the changes in files unrelated to your new backend.

**Comment by R11baka** [comment]
Hi ,@schachmat. You are welcome.
I have resolved conflict

File: .gitignore [hunk 1]
```diff
@@ -3,6 +3,7 @@
 *.a
 *.so
 /wego
+.idea/*
```
Comment (schachmat) [inline .gitignore]: remove

File: .gitignore [hunk 2]
```diff
@@ -27,3 +28,4 @@ _testmain.go
 # vim temp files
 *~
 *.swp
+Будет удалено .idea/
```
Comment (schachmat) [inline .gitignore]: remove

File: backends/forecast.io.go [hunk 3]
```diff
@@ -226,7 +226,7 @@ func (c *forecastConfig) Fetch(location string, numdays int) iface.Data {
 	if len(c.apiKey) == 0 {
 		log.Fatal("No forecast.io API key specified.\nYou have to register for one at https://developer.forecast.io/register")
 	}
-	if matched, err := regexp.MatchString(`^-?[0-9]*(\.[0-9]+)?,-?[0-9]*(\.[0-9]+)?$`, location); !matched || err != nil {
+	if matched, err := regexp.MatchString(`^-?[0-9]*(\.[0-9]+)?,\s*-?[0-9]*(\.[0-9]+)?$`, location); !matched || err != nil {
```
Comment (schachmat) [inline backends/forecast.io.go]: remove

File: backends/openweather.go [hunk 4]
```diff
@@ -0,0 +1,200 @@
+package backends
+
+import (
+	"flag"
+	//"fmt"
```
Comment (schachmat) [inline backends/openweather.go]: remove unused imports.

File: iface/iface.go [hunk 5]
```diff
@@ -142,7 +142,7 @@ func (u UnitSystem) Speed(spdKmph float32) (res float32, unit string) {
 }
 
 func (u UnitSystem) Distance(distM float32) (res float32, unit string) {
-	if u == UnitsMetric || u == UnitsSi || u == UnitsMetricMs{
+	if u == UnitsMetric || u == UnitsSi || u == UnitsMetricMs {
```
Comment (schachmat) [inline iface/iface.go]: should be a separate change, not part of this PR.

File: main.go [hunk 6]
```diff
@@ -33,6 +33,7 @@ func pluginLists() {
 }
 
 func main() {
+	log.SetOutput(os.Stdout)
```
Comment (schachmat) [inline main.go]: remove. we want to log to stderr by default.

File: main.go [hunk 7]
```diff
@@ -46,7 +47,7 @@ func main() {
 	flag.StringVar(location, "l", "40.748,-73.985", "`LOCATION` to be queried (shorthand)")
 	numdays := flag.Int("days", 3, "`NUMBER` of days of weather forecast to be displayed")
 	flag.IntVar(numdays, "d", 3, "`NUMBER` of days of weather forecast to be displayed (shorthand)")
-	unitSystem := flag.String("units", "metric", "`UNITSYSTEM` to use for output.\n    \tChoices are: metric, imperial, si")
+	unitSystem := flag.String("units", "metric", "`UNITSYSTEM` to use for output.\n    \tChoices are: metric, imperial, metric-ms, si")
```
Comment (schachmat) [inline main.go]: also unrelated to the new backend.

File: backends/openweather.go [hunk 8]
```diff
@@ -0,0 +1,200 @@
...
+	var ret iface.Data
+	s := strings.Split(location, ",")
+
+	lat, lon := s[0], s[1]
+
+	resp, err := ow.fetch(fmt.Sprintf(openweatherUri, lat, lon, ow.apiKey, ow.lang))
+	if err != nil {
+		log.Println("Cant fetch today")
+	}
+	ret.Current, err = ow.parseCond(resp.List[0])
```
Comment (schachmat) [inline backends/openweather.go]: This fails if the `resp` return is nil. Use `log.Fatalf()` previously (also including the returned `err`) to abort the program before trying to access `resp.List`.

**Comment by R11baka** [comment]
@schachmat  I have fixed changes

**Comment by R11baka** [comment]
@schachmat  I have fixed changes

**Comment by schachmat** [comment]
Thanks! I merged your PR and added some cleanup afterwards.
